Eliminated variants without accepted evidence

LeiChen0218 opened the pull request:
Fixed #19.  @gatoravi @malachig

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/griffithlab/civic-api-client/20)

<!-- Reviewable:end -->

LeiChen0218 committed:
eliminated variants without accepted evidence

gatoravi commented on civic_api_client/variants_lister.py:
```diff
@@ -377,12 +377,27 @@ def get_variant_details(self, variant_id):
+                        valid = True
+        return valid
+
+
```
This could also be written as so,

``` python
+    def evi_valid(self,variant_detail):
 +        "Fliter variant with no accepted evidence item"
 +        valid = False
 +        if 'evidence_items' in variant_detail:
 +            for evi in variant_detail['evidence_items']:
 +                if evi['status'] ==  "accepted":
 +                    valid = True
 +        return valid
```

LeiChen0218 replied:
Yes we can put it in valid test.
While since it takes a lot of time to run variant-lister, the way I wrote it is out of the consideration of saving time, which means if there's no valid evidence then we just pass this variant, don't even parse the variant details and do other test. 
@gatoravi

gatoravi replied:
Hi @LeiChen0218 yeah that definitely makes sense, I guess we could leave it as such and move it there in case there's more such 'priority-checks' that are added!

LeiChen0218 replied:
Yeah. And it might also be good to change "evi_valid" to "var_valid" or "prior_check"? @gatoravi

gatoravi replied:
Yeah sounds good, I like "prior_check" but either works!

gatoravi commented on civic_api_client/variants_lister.py:
```diff
@@ -377,12 +377,27 @@ def get_variant_details(self, variant_id):
+            if self.evi_valid(variant_details):
+                vd1 = VariantDetails(self.args, variant_details)
+                if vd1.satisfies_filters():
+                    self.filtered_variant_details.append(vd1)
```
Does it make sense for `evi_valid()` to be a part of `satisfies_filters()` method?

gatoravi commented:
We should also update the README to reflect this.

LeiChen0218 committed:
Merge remote-tracking branch 'origin/master' into vl-valid-filter

LeiChen0218 committed:
Add more usage instructions

LeiChen0218 commented:
Have changed the name of function and simplified the code. @gatoravi 
README was also updated.
(Please ignore the first commit, it was generated while I did fetch and merge...not a good reflection of the change)

gatoravi commented on README.md:
````diff
@@ -27,26 +27,61 @@ deactivate
+```
+4. Export all evidence items as tsv file 
+```
+civic-api-client tsv-creat
````
should be `tsv-create` here and in the code :space_invader:

gatoravi commented on civic_api_client/variants_lister.py:
```diff
@@ -377,12 +377,24 @@ def get_variant_details(self, variant_id):
 
+    def prior_check(self,variant_detail):
+        """Prior checks before parsing variant details:
+        1. Fliter variant with no accepted evidence item"""
```
Filter instead of Fliter

LeiChen0218 replied:
Should be fixed now. 
Special thanks and apologies to @gatoravi  and @malachig  for spending so much time on my terrible spelling every day lol.

gatoravi commented:
Looks great, just a couple of small typos. Should be ready to merge I think, once @malachig  gives the go-ahead! I'll try to run this as well.

LeiChen0218 committed:
Typos fixed

malachig commented:
This seems to be working now.  +1
